server/src/frameworks/Ape/ApeProject.ts

```diff
@@ -0,0 +1,233 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+
+import { existsSync, readFileSync } from "fs";
+import _ from "lodash";
+import { homedir } from "os";
+import path from "path";
+import {
+  CompletionItem,
+  DidChangeWatchedFilesParams,
+  Position,
+} from "vscode-languageserver-protocol";
+import YAML from "yaml";
+import { OpenDocuments, ServerState } from "../../types";
+import { toUnixStyle } from "../../utils";
+import { directoryContains } from "../../utils/directoryContains";
+import { CompilationDetails } from "../base/CompilationDetails";
+import { InitializationFailedError } from "../base/Errors";
+import { FileBelongsResult, Project } from "../base/Project";
+import { parseRemappingLine, Remapping } from "../base/Remapping";
+import { buildBasicCompilation } from "../shared/buildBasicCompilation";
+import { ApeConfig, ApeDependency } from "./types";
+
+export class ApeProject extends Project {
+  public priority = 1;
+  public sourcesPath!: string;
+  public dependenciesPath!: string;
+  public remappings: Remapping[] = [];
+  public initializeError?: string;
+  public configSolcVersion?: string;
+  public dependencies!: ApeDependency[];
+
+  constructor(
+    serverState: ServerState,
+    basePath: string,
+    public configPath: string
+  ) {
+    super(serverState, basePath);
+  }
+
+  public id(): string {
+    return this.configPath;
+  }
+
+  public frameworkName(): string {
+    return "Ape";
+  }
+
+  public async initialize(): Promise<void> {
+    this.initializeError = undefined; // clear any potential error on restart
+
+    try {
+      const configContent = readFileSync(this.configPath).toString();
+      const config: ApeConfig = YAML.parse(configContent);
+
+      this.configSolcVersion = config.solidity?.version;
+
+      const configSourcesPath = (
+        config.contracts_folder ?? "contracts"
+      ).replace("~", homedir());
```

Comment:
Is there a way of doing this with `path` manipulations?

Reply:
I used this trick because `path.resolve` doesn't handle the tilde. And from [ape's configuration docs](https://docs.apeworx.io/ape/stable/userguides/config.html), under `Contracts Folder`, it seems to support absolute paths using tildes.